### DISCUSSION TITLE
Set departure time in route planning

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#606] [Ideas]
Is there interest to set the departure date / time in route planning?

And automatically display the date / time at the stop points of the route.

This can be combined with the existing road book and future stop duration in via points:
- https://github.com/devemux86/cruiser/discussions/473

- https://github.com/devemux86/cruiser/discussions/573

<img src="https://github.com/devemux86/cruiser/assets/3484020/674fdfc6-4578-4293-8999-42f88c68e5ea" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/856f7052-4e7c-4a03-8dbe-1bcf8ad8bc0e" width="50%"/>

**Comment by GuG1050** [comment]
I guess this would only work for one-day-routes but not for several days in one route ?

**Reply by devemux86** [reply]
@GuG1050 the duration of the route should not really matter?

Setting the departure date / time, the arrival date / time of all stop points on the route can be calculated automatically.
Either on the same day or a few days later for really long routes spanning too many hours.

Like the above 1st image displaying arrival times in real time in the navigation's road book.

**Reply by 0709wiwiwi** [reply]
KISS is to produce one single file for each single day ?
Add a recognisable marker in each file name.   D01 D02 D03 etc ?

**Reply by devemux86** [reply]
@0709wiwiwi this is not related to route formats, that only store length / duration.
This is about the route on the map.

Arrival time is handled by the apps based on user's adjustable selection.
Like in navigation where is done based on real time.

**Reply by WolfgangZ42** [reply]
> but not for several days in one route ?

I never use multi day routes, but if I could define departure date/time (or stop duration) for a via point there shouldn't be any problem.

**Reply by devemux86** [reply]
@WolfgangZ42 

Do you mean define departure date / time not only on start point, but also on any via point?

Stop duration in via points can be combined here, but can also work as a separate function:
- https://github.com/devemux86/cruiser/discussions/573

**Reply by WolfgangZ42** [reply]
Yes, defining either stop duration or start date/time at any via point could give the ability to handle overnight stops as well.
